backend/gateway-service/src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Inject, Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import { firstValueFrom } from 'rxjs';
+import { LogInDto, SignUpDto, ValidateUserCredDto } from './dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @Inject('AUTH_SERVICE') private readonly authClient: ClientProxy,
+    @Inject('USER_SERVICE') private readonly userClient: ClientProxy,
+  ) {}
+
+  async signUp(data: SignUpDto) {
+    try {
+      const newUser = await firstValueFrom(
+        this.userClient.send({ cmd: 'create-user' }, data),
+      );
+
+      const token = await firstValueFrom(
+        this.authClient.send({ cmd: 'generate-jwt' }, newUser),
+      );
+
+      return {
+        token: token,
+        user: newUser,
+      };
+    } catch (error) {
+      throw new HttpException(
+        error.message || 'Error creating user',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
+
+  async logIn(data: LogInDto) {
+    try {
+      const user = await firstValueFrom(
+        this.userClient.send(
+          { cmd: 'get-user-by-email' },
+          { email: data.email },
+        ),
+      );
+
+      if (!user) {
+        throw new HttpException('User not found', HttpStatus.NOT_FOUND);
+      }
+
+      const payload: ValidateUserCredDto = {
+        password: data.password,
+        hashedPassword: user.password,
+      };
+      const isUserValid = await firstValueFrom(
+        this.authClient.send({ cmd: 'validate-user-cred' }, payload),
+      );
+      if (!isUserValid) {
+        throw new HttpException('Invalid credentials', HttpStatus.UNAUTHORIZED);
+      }
+
+      user.password = undefined;
```

Comment:
Would like clarify on the purpose of assigning the `undefined` value to user object

Reply:
It might be better to declare a partial DTO (e.g returning User object but without password attribute) for clarity